### DISCUSSION TITLE
Convert Posts to Pages - Open Source Utility - RETIRED

### DISCUSSION THREAD

**Discussion by bjazmoore** [GetPublii/Publii#1718] [Useful Resources] [answered]
### **THIS UTILTIY HAS BEEN RETIRED - USE NATIVE CONVERSION IN PUBLII 0.46.2**

**What is it:**  
A utility that converts Posts to Pages in Publii.

**Why:**
There have been a couple folks that have suggested an enhancement for Publii (and I am sure it is in the works) that will allow easy conversion of Posts to Pages.  
Also, since the concept of the page is really new, yet we have been creating pseudo pages from posts for a long time in Publii there is a need to migrate these pseudo pages to be real pages.

**What it does:**
There is a bit more to making a Post a page than just adding a status to the posts table.  This utility does all the extra stuff too - including making sure your local theme is ready for pages, creating (or adding) to the pages.config.json hierarchy file.  Migrating the postsViewSettings to be pageViewSettings.  Removing tags so that there are no orphans.

[**Learn more here**](https://github.com/bjazmoore/Post2Page-Converter)
[**Download it here**](https://github.com/bjazmoore/Post2Page-Converter/releases)

**Warning** - I developed this on Windows.  I have not tested it on platforms I do not use such as MacOS or Linux.

**Warning** - Always back up your site before migrating posts to pages.

**Warning** - The error checking is not supper robust.  I am sure it would be easy to use this program in a manner I did not expect and get unusual results.  Let me know what they are.

**Comment by candidexmedia** [comment]
Marvellous!

**Comment by dziudek** [comment, marked as answer]
@bjazmoore @candidexmedia - FYI - I will add option to change posts into pages and reverse in v.0.46.2 or v.0.47 - it has few things to do to make everything working properly. But I can confirm that it will be a core feature soon :)

**Reply by bjazmoore** [reply]
> @bjazmoore @candidexmedia - FYI - I will add option to change posts into pages and reverse in v.0.46.2 or v.0.47 - it has few things to do to make everything working properly. But I can confirm that it will be a core feature soon :)

Fantastic.  I kind of figured that would become a core feature.  Thanks

**Comment by s3nnet** [comment]
Fantastic!

**Comment by raramuridesign** [comment]
@bjazmoore Thank you. Tested and it works well. Two comments
1. After converting one post to page, I had to close the application and start again to complete the rest. It was a small site but did the job. Would be great to have a multi-select option.
2. After the conversion the items did not show in the pages area. Closing Publii and opening again, solved this.

Again - thanks for putting this out there.
M

**Reply by bjazmoore** [reply]
> @bjazmoore Thank you. Tested and it works well. Two comments
> 
> 1. After converting one post to page, I had to close the application and start again to complete the rest. It was a small site but did the job. Would be great to have a multi-select option.
> 2. After the conversion the items did not show in the pages area. Closing Publii and opening again, solved this.
> 
> Again - thanks for putting this out there. M

The issues in item one is odd.  I have tested several times trying to replicate but cannot.  

The good news - this will be native functionality in future versions of Publii and I can retire this ugly tool.

Note that it is best to close Publii when migrating, but not required - however, as you noticed some odd things happen if you do not.  I think this is because Publii caches the Pages and Posts.

**Reply by raramuridesign** [reply]
@bjazmoore for now your "dirty app" works as intended and this saves time... ;-)

**Comment by bjazmoore** [comment]
This utility will be retired now that Version 0.46.2 supports conversion natively.  Thanks
